### PR TITLE
Add utility function to enforce hash prefix uniqueness

### LINKED ISSUE #299: asv does not check if hash prefix is unique
@mdboom I noticed that ```asv``` isn't checking if the short hash is a unique benchmark identifier while investigating #298. This is primarily an issue in ```asv compare``` as the code averages over all benchmarks it finds which means, when combined with the bug described in #298, the first column will contain the average result for *all commits that start with 'e'* and the second will have *all commits that start with '9'*.

Given that the prefix of hash implies nothing about the relationship between the content, I believe the appropriate action would be to raise an ```Exception``` if the prefix is not unique. I'm working on a patch to address this but wanted to confirm that this approach would be more appropriate.

### PR DESCRIPTION
**Pull request by qwhelan**
@mdboom @pv Resolves #299 by adding a utility function to resolve a hash prefix to the "full" hash.

### CODE REVIEW

**Comment by pv** [comment]
Was this closed accidentally?

**Comment by qwhelan** [comment]
@pv Yeah, I was trying to retrigger the Travis run as a bunch of web tests randomly failed.

**Comment by pv** [comment]
Yes, there's something strange with phantomjs on travis, and the conda environment creation also seem to fail quite often due to web connectivity issues...

**Comment by qwhelan** [comment]
@pv Looks like it's all passing now. It looks like Travis has some caching infrastructure that might help cut down on the network failures, which I'll try to look into this evening.

File: asv/results.py [hunk 1]
```diff
@@ -80,6 +81,34 @@ def get_existing_hashes(results):
     return hashes
 
 
+def get_result_hash_from_prefix(results, machine_name, commit_prefix):
+    """
+    Get the 8-char result commit identifier from a potentially shorter
```
Comment (mdboom) [inline asv/results.py]: Instead of 8-char, don't you mean "full"?

Comment (qwhelan) [inline asv/results.py]: @mdboom Nope, the hash length is explicitly truncated to 8-chars below

**Comment by pv** [comment]
LGTM.

**Pull request merged**